### PR TITLE
feat: improve commit frequency and strategy for replicator and certifier

### PR DESCRIPTION
**Pull request by gk-kindred**

### CODE REVIEW

File: packages/cohort/src/replicator/core.rs [hunk 1]
```diff
@@ -182,4 +182,15 @@ where
         }
         statemaps_batch
     }
+
+    pub(crate) async fn commit_till_last_installed(&mut self) {
+        if self.last_installing > 0 {
+            if let Some(last_installed) = self.suffix.get_last_installed(Some(self.last_installing)) {
```
Comment (fmarek-kindred) [inline packages/cohort/src/replicator/core.rs:188]: minor: `Some` as param reads a bit strange here.

File: packages/cohort/src/replicator/services/replicator_service.rs [hunk 2]
```diff
@@ -20,6 +20,7 @@ where
     M: MessageReciever<Message = ChannelMessage> + Send + Sync,
 {
     info!("Starting Replicator Service.... ");
+    let mut interval = tokio::time::interval(Duration::from_millis(10_000));
```
Comment (fmarek-kindred) [inline packages/cohort/src/replicator/services/replicator_service.rs:23]: Consider extracting into configuration property

File: packages/cohort/src/replicator2/cohort_replicator.rs [hunk 3]
```diff
@@ -77,7 +77,7 @@ where
 
     pub async fn commit(&mut self) {
         if let Some(offset) = self.suffix.find_commit_offset(self.latest_in_flight) {
-            if let Err(error) = self.receiver.commit(offset).await {
+            if let Err(error) = self.receiver.commit().await {
```
Comment (fmarek-kindred) [inline packages/cohort/src/replicator2/cohort_replicator.rs:80]: It looks like we are extracting offset from suffix into Some(offset) but then calling receiver.commit() without params. Is this intentional?

**Review by fmarek-kindred — APPROVED** [review]

File: packages/cohort/src/replicator2/cohort_replicator.rs (hunk 3 above)
Comment (gk-kindred) [reply packages/cohort/src/replicator2/cohort_replicator.rs:80]: Yes this is fine, as at the time of commit we were always using the one stored on `tpl (topic partition list)`

File: packages/cohort/src/replicator/services/replicator_service.rs (hunk 2 above)
Comment (gk-kindred) [reply packages/cohort/src/replicator/services/replicator_service.rs:23]: I will update this in the new PR I am working on parallel installs, as I need to add few more environment variables in that PR. Will do it in one go.

**Pull request merged**